GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -384,24 +384,20 @@ private void FillNextPullActionAsDefaultToolStripMenuItems()
                 }
             };
 
-            var pullActionMenuItems = toolStripButtonPull.DropDownItems.OfType<ToolStripMenuItem>()
-                                                                       .Where(tsmi => tsmi.Tag is AppSettings.PullAction);
-
-            foreach (var pullActionMenuItem in pullActionMenuItems)
-            {
-                var menuItem = new ToolStripMenuItem
+            var setDefaultPullActionDropDownItems = toolStripButtonPull.DropDownItems
+                .OfType<ToolStripMenuItem>()
+                .Where(tsmi => tsmi.Tag is AppSettings.PullAction)
+                .Select(tsmi => (ToolStripItem)new ToolStripMenuItem
                 {
-                    Name = $"{pullActionMenuItem.Name}SetDefault",
-                    Text = pullActionMenuItem.Text,
+                    Name = $"{tsmi.Name}SetDefault",
+                    Text = tsmi.Text,
                     CheckOnClick = true,
-                    Image = pullActionMenuItem.Image,
-                    Tag = pullActionMenuItem.Tag
-                };
+                    Image = tsmi.Image,
+                    Tag = tsmi.Tag
+                })
+                .Touch(tsmi => tsmi.Click += SetDefaultPullActionMenuItemClick);
```

Comment:
is there a reason why we couldn't write the following:

```
.Where(...)
.Select(tsmi => 
{
    var newTsmi = new ToolStripMenuItem
    {
        Name = ....
    };
    newTsmi.Click += SetDefaultPullActionMenuItemClick;
    return newTsmi;
})
.ToArray();
```

this way there is no need for a new extension method (which otherwise needs to be tested and maintained)

Reply:
A couple of points
1. Touch is more pure/functional. You `map` in `Select` and you `visit` in `Touch`. Sure you can do all in `Select` but then you are not _just_ selecting (mapping)
2. It looks nicer :)

Also `Touch` is really really simple method. It's basically ForEach except it works in `IEnumerable<T>` chain.

These are all the arguments I've got :)

Reply:
So I had a discussion with a colleague about `.Touch` and now I am on an even shakier ground. Still like how it looks/feels but my arguments have less weight. I'll revert it if you want.